test/test_linalg.py

```diff
@@ -2845,203 +2772,54 @@ def run_subtest(actual_rank, matrix_size, batches, device, svd_lowrank, **option
-                svd(a, out=(out_u, out_s, out_v))
-                # Check warning occurs
-                self.assertEqual(len(w), 3)
-                self.assertTrue("An output with one or more elements was resized" in str(w[-3].message))
-                self.assertTrue("An output with one or more elements was resized" in str(w[-2].message))
-                self.assertTrue("An output with one or more elements was resized" in str(w[-1].message))
-
-            # dtypes should be safely castable
-            out_u = torch.empty(0, dtype=torch.int, device=device)
-            out_s = torch.empty(0, dtype=torch.int, device=device)
-            out_v = torch.empty(0, dtype=torch.int, device=device)
-            with self.assertRaisesRegex(RuntimeError, "but got U with dtype Int"):
-                svd(a, out=(out_u, out_s, out_v))
-
-            out_u = torch.empty(0, dtype=dtype, device=device)
-            if svd == torch.linalg.svd:
-                msg = "but got Vh with dtype Int"
-            else:
-                msg = "but got V with dtype Int"
-            with self.assertRaisesRegex(RuntimeError, msg):
-                svd(a, out=(out_u, out_s, out_v))
-
-            out_v = torch.empty(0, dtype=dtype, device=device)
-            with self.assertRaisesRegex(RuntimeError, "but got S with dtype Int"):
-                svd(a, out=(out_u, out_s, out_v))
-
-            # device should match
-            if torch.cuda.is_available():
-                wrong_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
-                out_u = torch.empty(0, device=wrong_device, dtype=dtype)
-                out_s = torch.empty(0, device=wrong_device, dtype=real_dtype)
-                out_v = torch.empty(0, device=wrong_device, dtype=dtype)
-                with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
-                    # error from out_u
-                    svd(a, out=(out_u, out_s, out_v))
+    def test_svd(self, device, dtype):
+        # tests linalg.svd, svd, linalg.svdvals
+        make_arg = partial(make_tensor, dtype=dtype, device=device)
 
-                out_u = torch.empty(0, device=device, dtype=dtype)
-                with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
-                    # error from out_s
-                    svd(a, out=(out_u, out_s, out_v))
+        backends = ["default"]
 
-                out_s = torch.empty(0, device=device, dtype=real_dtype)
-                with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
-                    # error from out_v
-                    svd(a, out=(out_u, out_s, out_v))
-
-            # if input contains NaN then an error is triggered for svd
-            # When cuda < 11.5, cusolver raises CUSOLVER_STATUS_EXECUTION_FAILED when input contains nan.
-            # When cuda >= 11.5, cusolver normally finishes execution and sets info array indicating convergence issue.
```

Comment:
Was this part removed on purpose?

Reply:
I removed the whole test becaues it mostly tested things that the OpInfo tests. I think this got in in one of the rebases, sorry. Let's cherry-pick the things that it'd be good to test and let's write a new comprehensive and compact test similar to the new `test_svd` that I wrote in this PR. What do you think?

Reply:
I copied this particular part to https://github.com/pytorch/pytorch/pull/68812.